build_projects/dotnet-cli-build/MakeRelative.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.IO;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class MakeRelative : Task
+    {
+        [Required]
+        public string Path1 { get; set; }
+
+        [Required]
+        public string Path2 { get; set; }
+
+        public char SeparatorChar { get; set; }
+
+        [Output]
+        public ITaskItem RelativePath { get; set; }
+
+        public override bool Execute()
+        {
+            if (SeparatorChar == default(char))
+            {
+                SeparatorChar = Path.DirectorySeparatorChar;
+            }
+
+            var relativePath = GetRelativePath(Path1, Path2, SeparatorChar);
+
+            RelativePath = ToTaskItem(Path1, Path2, relativePath);
+
+            return true;
+        }
+
+        private static TaskItem ToTaskItem(string path1, string path2, string relativePath)
+        {
+            var framework = new TaskItem();
+            framework.ItemSpec = relativePath;
+
+            framework.SetMetadata("Path1", path1);
+            framework.SetMetadata("Path2", path2);
+            framework.SetMetadata("RelativePath", relativePath);
+
+            return framework;
+        }
+
+        private static string GetRelativePath(string path1, string path2, char separator = default(char))
```

Comment:
Can we just compile Include the existing file that contains this logic so it isn't duped another time?
[https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Cli.Utils/PathUtility.cs#L94](https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Cli.Utils/PathUtility.cs#L94)

I really dislike duping code, especially in the same repo.